conjure-java-core/src/main/java/com/palantir/conjure/java/util/SafetyUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.conjure.java.types.SafetyEvaluator;
+import com.palantir.conjure.spec.AliasDefinition;
+import com.palantir.conjure.spec.ArgumentDefinition;
+import com.palantir.conjure.spec.ExternalReference;
+import com.palantir.conjure.spec.FieldDefinition;
+import com.palantir.conjure.spec.ListType;
+import com.palantir.conjure.spec.LogSafety;
+import com.palantir.conjure.spec.MapType;
+import com.palantir.conjure.spec.OptionalType;
+import com.palantir.conjure.spec.PrimitiveType;
+import com.palantir.conjure.spec.SetType;
+import com.palantir.conjure.spec.Type;
+import com.palantir.conjure.spec.TypeName;
+import java.util.Optional;
+
+public final class SafetyUtils {
+    private SafetyUtils() {}
+
+    // SafetyEvaluator uses its TypeDefinitionMap to resolve the safety of reference types, but these utils
+    // only evaluate the safety of external imports and primitives, so it's safe to use a placeholder
+    private static final SafetyEvaluator safetyEvaluator = new SafetyEvaluator(ImmutableMap.of());
```

Comment:
As al alternative, we could make these methods non-static and add them to the SafetyEvaluator -- that may be more robust (or simply add the static methods to SafetyEvaluator so the safety evaluation pieces live in the same place)

Reply:
I was trying to avoid plumbing through a SafetyEvaluator instance everywhere these utils are used, but it's probably a good idea to do so

Reply:
I think we already have one plumbed in a bunch of places where it's not actually used, heh. If that turns out to require a great deal of plumbing, we could use static methods on SafetyEvaluator, and defer the plumbing to a follow-up commit. Up to you.